ledger/ledger-api-common/src/main/scala/com/digitalasset/platform/server/api/validation/FieldValidations.scala

```diff
@@ -144,6 +144,36 @@ object FieldValidations {
   ): Either[StatusRuntimeException, Ref.LedgerString] =
     requireNonEmptyParsedId(Ref.LedgerString.fromString)(s, fieldName)
 
+  def optionalLedgerString(
+      s: String,
+      fieldName: String,
+  )(implicit
+      contextualizedErrorLogger: ContextualizedErrorLogger
+  ): Either[StatusRuntimeException, Option[Ref.LedgerString]] =
+    optionalString(s) { nes => requireLedgerString(nes, fieldName) }
+
+  def eventSequentialId(raw: String, fieldName: String)(implicit
+      errorLogger: ContextualizedErrorLogger
+  ): Either[StatusRuntimeException, Long] = {
+    Try {
+      raw.toLong
+    } match {
+      case Success(seqId) => Right(seqId)
+      case Failure(_) =>
+        // Do not mention event sequential id as this should be opaque externally
+        Left(invalidField(fieldName = fieldName, message = "Invalid field value"))
+    }
+  }
+
+  def optionalEventSequentialId(
```

Comment:
now that it's opaque, this name no longer matches the semantic of the field

Reply:
Message is now provided as parameter.